src/credential/sd_jwt.rs

```diff
@@ -0,0 +1,364 @@
+    pub(crate) inner: SdJwtBuf,
+}
+
+// Internal utility methods for decoding a SdJwt.
+impl SdJwt {
+    /// Decode a SdJwt instance and return the revealed claims as a JSON value.
+    pub fn decode_reveal_json(&self) -> Result<serde_json::Value, SdJwtError> {
+        serde_json::to_value(&self.revealed_claims)
+            .map_err(|e| SdJwtError::Serialization(e.to_string()))
+    }
+
+    /// The types of the credential from the VCDM, excluding the base `VerifiableCredential` type.
+    pub fn types(&self) -> Vec<String> {
+        self.credential.additional_types().to_vec()
+    }
+
+    /// Returns the SD-JWT credential as an AnyCredential type.
+    pub fn credential(&self) -> AnyJsonCredential {
+        AnyJsonCredential::V2(self.credential.clone())
+    }
+
+    /// Check if the credential satisfies a presentation definition.
+    pub fn check_presentation_definition(&self, definition: &PresentationDefinition) -> bool {
+        // If the credential does not match the definition requested format,
+        // then return false.
+        if !definition
+            .format()
+            .contains_key(&ClaimFormatDesignation::Other("vcdm2_sd_jwt".into()))
+        {
+            return false;
+        }
+
+        let Ok(json) = serde_json::to_value(&self.credential) else {
+            // NOTE: if we cannot convert the credential to a JSON value, then we cannot
+            // check the presentation definition, so we return false.
+            //
+            // TODO: add logging to indicate that the credential could not be converted to JSON.
+            return false;
+        };
+
+        // Check the JSON-encoded credential against the definition.
+        definition.check_credential_validation(&json)
+    }
+}
+
+#[uniffi::export]
+impl SdJwt {
+    /// Create a new SdJwt instance from a compact SD-JWS string.
+    #[uniffi::constructor]
+    pub fn new_from_compact_sd_jwt(input: String) -> Result<Arc<Self>, SdJwtError> {
+        let inner: SdJwtBuf =
+            SdJwtBuf::new(input).map_err(|e| SdJwtError::InvalidSdJwt(e.to_string()))?;
+
+        let mut sd_jwt = SdJwt::try_from(inner)?;
+        sd_jwt.key_alias = None;
+
+        Ok(Arc::new(sd_jwt))
+    }
+
+    /// Create a new SdJwt instance from a compact SD-JWS string with a provided key alias.
+    #[uniffi::constructor]
+    pub fn new_from_compact_sd_jwt_with_key(
+        input: String,
+        key_alias: KeyAlias,
+    ) -> Result<Arc<Self>, SdJwtError> {
+        let inner: SdJwtBuf =
+            SdJwtBuf::new(input).map_err(|e| SdJwtError::InvalidSdJwt(e.to_string()))?;
+
+        let mut sd_jwt = SdJwt::try_from(inner)?;
+        sd_jwt.key_alias = Some(key_alias);
+
+        Ok(Arc::new(sd_jwt))
+    }
+
+    /// Create an instance of a SdJwt from a generic credential type.
+    #[uniffi::constructor]
+    pub fn new_from_credential(credential: &Credential) -> Result<Arc<Self>, SdJwtError> {
+        Self::try_from(credential).map(Arc::new)
+    }
+
```

Comment:
Can you remove this from the uniffi-exported block? The expectation is that a consumer of the Rust code will never construct a "new" SdJwt from a Credential. For that the consumer should go through the ParsedCredential interface.